nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ChangeNodeMasterPasswordCommand.java

```diff
@@ -171,89 +164,88 @@ public int execute(String... argv) throws CommandException {
     }
 
     /**
-     * Create the master password keystore. This routine can also modify the master password
-     * if the keystore already exists
-     *
-     * @throws CommandException
+     * Find the old password from the property in the password file with the name
+     * {@link #OLD_PASSWORD_ALIAS} if it exists, or by prompting the user otherwise.
+     * 
+     * @throws CommandException if the password is null
      */
-    protected void createMasterPasswordFile() throws CommandException {
-        final File pwdFile = new File(this.getServerDirs().getAgentDir(), MASTER_PASSWORD_ALIAS);
-        try {
-            PasswordAdapter p = new PasswordAdapter(pwdFile.getAbsolutePath(),
-                MASTER_PASSWORD_ALIAS.toCharArray());
-            p.setPasswordForAlias(MASTER_PASSWORD_ALIAS, newPassword.getBytes());
-            FileProtectionUtility.chmod0600(pwdFile);
-        } catch (Exception ex) {
-            throw new CommandException(strings.get("masterPasswordFileNotCreated", pwdFile),ex);
+    protected String findOldPassword() throws CommandException {
+        // Fetch from password file
+        String oldPassword = passwords.get(OLD_PASSWORD_ALIAS);
+
+        // Prompt user
+        if (oldPassword == null) {
+            char[] opArr = super.readPassword(STRINGS.get("old.mp"));
```

Comment:
This (or possibly the whole command) probably needs to be hidden behind an `isInteractive()` check.
Currently if you run `asadmin -I false change-master-password Noddy1` it will still prompt you for a password.

Reply:
Good shout, the `getPassword` method did this but the `readPassword` method didn't. I've added the check